src/redux/actions/assetRegisterApi.js

```diff
@@ -12,6 +13,104 @@ export const ASSET_GET_REQUEST = Symbol('ASSET_GET_REQUEST');
 export const ASSET_GET_SUCCESS = Symbol('ASSET_GET_SUCCESS');
 export const ASSET_GET_FAILURE = Symbol('ASSET_GET_FAILURE');
 
+// Shared fields between FILTER_FIELDS and SORT_FIELDS.
+const COMMON_FIELDS = [
+  'id', 'name', 'department', 'purpose', 'owner', 'private', 'research', 'personal_data',
+  'data_subject', 'data_category', 'recipients_category', 'recipients_outside_eea', 'retention',
+  'risk_type', 'storage_location', 'storage_format', 'paper_storage_security', 'is_complete',
+  'digital_storage_security',
+];
+
+// Set of fields which can be used as a filter
+export const FILTER_FIELDS = new Set(COMMON_FIELDS);
+
+// Set of fields which can be used for sorting
+export const SORT_FIELDS = new Set([...COMMON_FIELDS, 'created_at', 'updated_at']);
+
+export const Direction = Object.freeze({
+  ascending: Symbol('ascending'), descending: Symbol('descending')
+});
+
+/**
+ * Fetch a new set of assets. Optionally passed an object containing the following properties:
+ *
+ * sort: (optional) object with following structure:
+ *
+ *  { field, direction }
+ *
+ *    field: (optional) String giving the field name which should be used for ordering the returned
+ *      results. By default the results are ordered by created_at. Must be one of the strings in
+ *      SORT_FIELDS. If sortBy is not in SORT_FIELDS, this is ignored. Default: created_at.
+ *
+ *    direction: (optional) One of Direction.ascending or Direction.descending depending on what
+ *      order the sorted results should be returned in. Invalid values are ignored. Default:
+ *      Direction.descending.
+ *
+ * search: (optional) String which should be matched against all asset fields.
+ *
+ * filter: (optional) Object whose properties specify filters. Only the properties in FILTER_FIELDS
+ *   are looked at. Other properties are ignored.
+ *
+ */
+export const getAssets = ({
+  sort: { field: sortBy = 'created_at', direction = Direction.descending } = { }, search, filter
+} = {}) => {
+  // Build a query string from the options passed to this action.
+
+  // Array of [key, value] pairs which are built into the query string.
+  const queryParts = [];
+
+  // Object representing this query in the state.
+  const query = { sort: { field: null, direction: null }, filter: { }, search: null };
```

Comment:
would it be simpler to pass query as parameter? - then you would not need to reconstruct

Reply:
Yes, yes it would. Not quite sure what I was thinking here.

Reply:
I remember now: the query is re-built from the input so that extra or invalid fields are not passed through to the action.

Reply:
I'll re-work the function to make this clearer.